edu-services/cm-service/cm-impl/hibernate-impl/impl/src/java/org/sakaiproject/coursemanagement/impl/CourseManagementServiceHibernateImpl.java

```diff
@@ -389,11 +389,25 @@ public boolean isEmpty(final String courseSetEid) {
 	public Map<String, String> findSectionRoles(final String userEid) {
 		List results = getHibernateTemplate().findByNamedQueryAndNamedParam(
 				"findSectionRoles", "userEid", userEid);
-		Map<String, String> sectionRoleMap = new HashMap<String, String>();
-		for(Iterator iter = results.iterator(); iter.hasNext();) {
-			Object[] oa = (Object[])iter.next();
+		return createSectionRolesMapFromQueryResults(results);
+	}
+
+	public Map<String, String> findSectionRolesMatchingRoles(final String userEid, final List<String> roles) {
+		List results = getHibernateTemplate().findByNamedQueryAndNamedParam(
+				"findSectionRolesMatchingRoles", new String[]{"userId", "roles"}, new Object[]{userEid, roles});
+		return createSectionRolesMapFromQueryResults(results);
+	}
+
+	/**
+	 * findSectionRoles queries return results as Object[][] in the form {[section1, role1], [section2, role2], ...}.
+	 * This method converts this into a map
+	 */
+	private Map<String, String> createSectionRolesMapFromQueryResults(List queryResults) {
+		final Map<String, String> sectionRoleMap = new HashMap<>();
```

Comment:
since your using a lambda here just go the extra step you can replace everything in that method with:
`return results.stream().flatMap(Arrays::stream).collect(Collectors.toMap(o -> (String) o[0], o -> (String) o[1]));`